app/src/main/java/org/oppia/app/player/state/InteractionAdapter.kt

```diff
@@ -0,0 +1,160 @@
+package org.oppia.app.player.state
+
+import android.text.Spannable
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.databinding.library.baseAdapters.BR
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.item_selection_interaction_items.view.*
+import kotlinx.android.synthetic.main.multiple_choice_interaction_items.view.*
+import org.oppia.app.R
+import org.oppia.app.databinding.ItemSelectionInteractionItemsBinding
+import org.oppia.app.databinding.MultipleChoiceInteractionItemsBinding
+import org.oppia.app.model.InteractionObject
+import org.oppia.app.model.StringList
+import org.oppia.app.player.state.itemviewmodel.CustomizationArgsInteractionViewModel
+import org.oppia.app.player.state.itemviewmodel.SelectionContentViewModel
+import org.oppia.app.player.state.listener.InteractionAnswerRetriever
+import org.oppia.util.parser.HtmlParser
+
+private const val VIEW_TYPE_MULTIPLE_CHOICE = 1
+private const val VIEW_TYPE_ITEM_SELECTION = 2
+private const val INTERACTION_ADAPTER_TAG = "Interaction Adapter"
+
+/** Adapter to bind the interactions to the [RecyclerView]. It handles MultipleChoiceInput and ItemSelectionInput interaction views. */
```

Comment:
Please wrap this if it exceeds the line limit.

Reply:
done.